applications/nrf_desktop/README.rst

```diff
@@ -1766,53 +1776,109 @@ The B0 bootloader requires the following options enabled:
+
+It is recommended to also enable the :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_REJECT_DIRECT_XIP_MISMATCHED_SLOT` in the application configuration to make sure that MCUmgr will reject application image updates with invalid start address.
+This prevents uploading an update image build for improper slot through the MCUmgr's Simple Management Protocol (SMP).
 
-Configure the MCUboot bootloader with the following options:
+Serial recovery mode
+~~~~~~~~~~~~~~~~~~~~
 
-* :kconfig:option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` - This option defines the path to the private key that is used to sign the application and that is used by the bootloader to verify the application signature.
-  The key must be defined only in the MCUboot bootloader configuration file.
-* :kconfig:option:`CONFIG_IMG_MANAGER` and :kconfig:option:`CONFIG_MCUBOOT_IMG_MANAGER` - These options allow the application to manage the DFU image.
-  Enable both of them only for configurations that support :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`.
-  For these configurations, the :ref:`nrf_desktop_dfu` uses the provided API to request firmware upgrade and confirm the running image.
+In the :ref:`USB serial recovery <nrf_desktop_bootloader_serial_dfu>` mode, the MCUboot bootloader uses a built-in foreground DFU transport over serial interface through USB.
+The application is not involved in the foreground DFU transport and therefore it can be directly overwritten by the bootloader.
+Because of that, the configuration with serial recovery mode requires only a single application slot.
+
+Enable the USB serial recovery DFU using the following configuration options:
+
+* ``CONFIG_MCUBOOT_SERIAL`` - This option enables the serial recovery DFU.
+* ``CONFIG_BOOT_SERIAL_CDC_ACM`` - This option enables the serial interface through USB.
+
+  .. note::
+    Make sure to enable and properly configure the USB subsystem in the bootloader configuration.
+    See :ref:`usb_api` for more information.
+
+The MCUboot bootloader enters serial recovery mode instead of booting the application if predefined button is pressed during boot.
+The GPIO pin used to trigger the serial recovery mode is configured using Devicetree Specification (DTS).
+The pin is configured with the ``mcuboot-button0`` alias.
+The ``mcuboot-led0`` alias can be used to define LED activated in the serial recovery mode.
+The ``CONFIG_MCUBOOT_INDICATION_LED`` Kconfig option must be selected to enable the LED.
+By default, both the GPIO pin and the LED are defined in board's DTS file.
+See :file:`boards/arm/nrf52833dongle_nrf52833/nrf52833dongle_nrf52833.dts` for an example of configuration.
+
+For an example of configuration, see the MCUboot bootloader debug configuration defined for nRF52833 dongle (:file:`applications/nrf_desktop/configuration/nrf52833dongle_nrf52833/child_image/mcuboot/prj.conf`).
```

Comment:
@MarekPieta Is there any way we can better differentiate lines 1847 and 1849? They both have very similar descriptions